rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -286,16 +317,26 @@ void MCAPStorage::open_impl(const std::string& uri,
       relative_path_ = uri + FILE_EXTENSION;
 
       mcap_writer_ = std::make_unique<mcap::McapWriter>();
-      McapWriterOptions options;
+      McapWriterOptions mcap_writer_options;
+      WriteBufferingOptions write_buffering_options;
       if (!storage_config_uri.empty()) {
         YAML::Node yaml_node = YAML::LoadFile(storage_config_uri);
-        options = yaml_node.as<McapWriterOptions>();
+        mcap_writer_options = yaml_node.as<McapWriterOptions>();
+        write_buffering_options = yaml_node.as<WriteBufferingOptions>();
       }
-
-      auto status = mcap_writer_->open(relative_path_, options);
-      if (!status.ok()) {
-        throw std::runtime_error(status.message);
+      sync_after_write_ = write_buffering_options.syncAfterWrite;
+      flush_after_write_ = write_buffering_options.bufferEntireBatch;
+      if (write_buffering_options.bufferEntireBatch) {
+        buffered_writer_->open(relative_path_, std::nullopt);
```

Comment:
The `open()` call below throws on `success=false` but this one does not. Is that intentional?